libraries/AP_Scripting/examples/copter-fast-descent.lua

```diff
@@ -0,0 +1,158 @@
+-- Copter descends very rapidly in a spiral pattern to a present altitude above home
+--
+-- CAUTION: This script only works for Copter
+-- this script waits for the vehicle to be changed to Guided mode and then:
+--    a) flies a spiral pattern using the velocity and acceleration control
+--    b) slows the spiral and stops at the preset altitude
+--    c) switches to RTL
+
+-- constants
+local copter_guided_mode_num = 4    -- Guided mode is 4 on copter
+local copter_rtl_mode_num = 6       -- RTL is 6 on copter
+local alt_above_home_min = 50       -- copter will stop at this altitude above home
+local circle_radius_rate_max_ms = 1 -- radius expands at max of this many m/s
+local circle_radius_accel_mss = 1   -- radius expansion speed accelerates at this many m/s/s
+local circle_radius_max = 30        -- target circle's maximum radius
+local speed_xy_max = 5              -- max target horizontal speed
```

Comment:
You could read this, and the max accels out of params. For just a one time read you could do 
```
local speed_xy_max = assert(param:get('WPNAV_SPEED'),'failed to get param') / 100
```

Or to make it update in real time.
```
local speed_xy_max = Parameter()
assert(speed_xy_max:init('WPNAV_SPEED'),'failed to get param') 
```
and call with
```
(speed_xy_max:get() / 100)
```

Reply:
I think I'll leave the horizontal speed separate from the wpnav-speed.  The reason is that the max horizontal speed depends upon the max lean angle of the vehicle and the radius of the circle.  It is relatively easy to have a situation where the vehicle can't lean over far enough to both stay on the circle and fly at the desired horizontal speed.  In normal flight a vehicle might be able to fly at 15m/s with a relatively low lean max lean angle (e.g. 25deg) but for these fast spirals it'll need to go much slower.  Of course there is probably a formula which could be added to ensure the max speed is not too high ..